### PR TITLE
ZEPPELIN-273 Spark 1.5 support

### PR DESCRIPTION
**Pull request by Leemoonsoo**
https://issues.apache.org/jira/browse/ZEPPELIN-273

"spark-1.5" profile is added

### CODE REVIEW

**Comment by Leemoonsoo** [comment]
With adding spark-1.5 profile, some duplicated codes are removed.
Some conditional statements that branching code based on spark version is simplified from comparing version numerical way.

Ready to review and merge

File: spark/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java [hunk 1]
```diff
@@ -286,9 +286,7 @@ public InterpreterResult interpret(String st, InterpreterContext context) {
     }
 
     SparkInterpreter sparkInterpreter = getSparkInterpreter();
-    if (!sparkInterpreter.getSparkContext().version().startsWith("1.2") &&
-        !sparkInterpreter.getSparkContext().version().startsWith("1.3") &&
-        !sparkInterpreter.getSparkContext().version().startsWith("1.4")) {
+    if (sparkInterpreter.getSparkVersion() < 120) {
```
Comment (djoelz) [inline spark/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java]: should we make 120 a class constant?

File: spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java [hunk 2]
```diff
@@ -453,15 +456,9 @@ public void open() {
                  + "_binder.get(\"sqlc\").asInstanceOf[org.apache.spark.sql.SQLContext]");
     intp.interpret("import org.apache.spark.SparkContext._");
 
-    if (sc.version().startsWith("1.1")) {
-      intp.interpret("import sqlContext._");
-    } else if (sc.version().startsWith("1.2")) {
+    if (sparkVersion < 130) {
```
Comment (djoelz) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: should we make 130 a class constant?

File: spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java [hunk 3]
```diff
@@ -479,14 +476,10 @@ public void open() {
      */
 
     try {
-      if (sc.version().startsWith("1.1") || sc.version().startsWith("1.2")) {
+      if (sparkVersion < 130) {
```
Comment (djoelz) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: should we make 130 a class constant?

File: spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java [hunk 4]
```diff
@@ -671,18 +664,10 @@ public int getProgress(InterpreterContext context) {
 
       if (jobGroup.equals(g)) {
         int[] progressInfo = null;
-        if (sc.version().startsWith("1.0")) {
+        if (sparkVersion < 110) {
```
Comment (djoelz) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: should we make 110 a class constant?

File: spark/src/main/resources/python/zeppelin_pyspark.py [hunk 5]
```diff
@@ -84,9 +84,9 @@ def get(self, key):
 sys.stderr = output
 
 client = GatewayClient(port=int(sys.argv[1]))
-sparkVersion = sys.argv[2]
+sparkVersion = int(sys.argv[2])
 
-if sparkVersion.startswith("1.4"):
+if sparkVersion >= 140:
```
Comment (djoelz) [inline spark/src/main/resources/python/zeppelin_pyspark.py]: Same feedback on constants

File: spark/src/main/resources/python/zeppelin_pyspark.py [hunk 6]
```diff
@@ -102,15 +102,12 @@ def get(self, key):
 
 jsc = intp.getJavaSparkContext()
 
-if sparkVersion.startswith("1.2"):
+if sparkVersion < 130:
```
Comment (djoelz) [inline spark/src/main/resources/python/zeppelin_pyspark.py]: Same feedback on constants

File: spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java [hunk 7]
```diff
@@ -803,4 +788,19 @@ public Scheduler getScheduler() {
...
+   * @param sc SparkContext
+   * @return integer number. eg) 141 for spark "1.4.1"
+   */
+  private int getVersion(SparkContext sc) {
+    String versionString = sc.version();
+    versionString = versionString.replaceAll("\\.", "");
+    return Integer.parseInt(versionString);
+  }
+
+  public int getSparkVersion() {
```
Comment (djoelz) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: unit tests please

**Comment by felixcheung** [comment]
Instead of int/const, should we use enum for this? Instead we could say:

`hasDataFrame` or `hasSchemaRDD` etc?

**Comment by Leemoonsoo** [comment]
@djoelz @felixcheung enum class and test added.

**Comment by djoelz** [comment]
LGTM

File: spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java [hunk 8]
```diff
@@ -671,18 +664,10 @@ public int getProgress(InterpreterContext context) {
 
       if (jobGroup.equals(g)) {
         int[] progressInfo = null;
-        if (sc.version().startsWith("1.0")) {
+        if (sparkVersion.olderThan(SparkVersion.SPARK_1_1_0)) {
```
Comment (felixcheung) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: nit: make this sparkVersion.getProgress1_0()?

File: spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java [hunk 9]
```diff
@@ -479,14 +476,10 @@ public void open() {
      */
 
     try {
-      if (sc.version().startsWith("1.1") || sc.version().startsWith("1.2")) {
+      if (sparkVersion.olderThan(SparkVersion.SPARK_1_3_0)) {
```
Comment (felixcheung) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: nit: make this sparkVersion.oldLoadFilesMethodName()

File: spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java [hunk 10]
```diff
@@ -453,15 +456,9 @@ public void open() {
                  + "_binder.get(\"sqlc\").asInstanceOf[org.apache.spark.sql.SQLContext]");
     intp.interpret("import org.apache.spark.SparkContext._");
 
-    if (sc.version().startsWith("1.1")) {
-      intp.interpret("import sqlContext._");
-    } else if (sc.version().startsWith("1.2")) {
+    if (sparkVersion.olderThan(SparkVersion.SPARK_1_3_0)) {
```
Comment (felixcheung) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: nit: make this sparkVersion.oldSqlContextImplicits()

**Comment by felixcheung** [comment]
looks good! thanks for this, I'm going to pick this up with the "provided spark" change

**Comment by Leemoonsoo** [comment]
@djoelz @felixcheung Thanks for good feedbacks.

**Comment by felixcheung** [comment]
awesome!

**Comment by Leemoonsoo** [comment]
merging if there is no more discussions

**Comment by bzz** [comment]
Looks great to me :shipit:

**Pull request closed without merging**